pkg/combustion/kubernetes.go

```diff
@@ -114,16 +121,23 @@ func configureRKE2(ctx *image.Context) (string, error) {
 		return "", fmt.Errorf("downloading RKE2 artefacts: %w", err)
 	}
 
+	manifestsPath := ""
+	if isComponentConfigured(ctx, filepath.Join(k8sDir, "manifests")) {
+		manifestsPath = manifestsDir
+	}
+
 	rke2 := struct {
 		image.Kubernetes
-		ConfigFile  string
-		InstallPath string
-		ImagesPath  string
+		ConfigFile    string
+		InstallPath   string
+		ImagesPath    string
+		ManifestsPath string
 	}{
-		Kubernetes:  ctx.ImageDefinition.Kubernetes,
-		ConfigFile:  configFile,
-		InstallPath: installPath,
-		ImagesPath:  imagesPath,
+		Kubernetes:    ctx.ImageDefinition.Kubernetes,
+		ConfigFile:    configFile,
+		InstallPath:   installPath,
+		ImagesPath:    imagesPath,
+		ManifestsPath: manifestsPath,
```

Comment:
It is very weird to render the template with the directory and only after that actually attempt to `configureManifests()`. The more I think about it, the more I believe that `configureManifests()` should also be called as part of this function instead of using `configureKubernetes()` for them.

If the concern here is that the same would apply for k3s installations, I'd rather leave that for me once I get to introducing those changes. In the meantime, it's best not to mix up the internal state - `manifestsPath` and parsing the template should only be done if `configureManifests()` is successful.